ios/FluentUI/Vnext/HUD/HeadsUpDisplay.swift

```diff
@@ -0,0 +1,227 @@
+        let type = state.type
+        let foregroundColor = Color(dynamicColor: tokens.foregroundColor)
+        let verticalPadding = tokens.verticalPadding
+        let horizontalPadding = tokens.horizontalPadding
+
+        HStack(alignment: .center) {
+            VStack {
+                switch type {
+                case .activity:
+                    ActivityIndicator(size: .xLarge)
+                        .isAnimating(true)
+                        .color(UIColor(dynamicColor: tokens.foregroundColor))
+                case .custom, .failure, .success:
+                    let image: UIImage = {
+                        switch type {
+                        case.activity:
+                            preconditionFailure("HUDType.activity does not have an associated image.")
+                        case .custom(let image):
+                            return image
+                        case .failure:
+                            return UIImage.staticImageNamed("dismiss-36x36")!
+                        case.success:
+                            return UIImage.staticImageNamed("checkmark-36x36")!
+                        }
+                    }()
+
+                    Image(uiImage: image)
+                        .foregroundColor(foregroundColor)
+                }
+
+                if !label.isEmpty {
+                    Spacer()
+                        .frame(height: verticalPadding)
+                    Text(label)
+                        .foregroundColor(foregroundColor)
+                        .lineLimit(2)
+                        .multilineTextAlignment(.center)
+                        .fixedSize(horizontal: false, vertical: true)
+                }
+            }
+        }
+        .padding(EdgeInsets(top: verticalPadding,
+                            leading: horizontalPadding,
+                            bottom: verticalPadding,
+                            trailing: horizontalPadding))
+        .squareShaped(minSize: tokens.minSize,
+                      maxSize: tokens.maxSize)
+        .background(Rectangle()
+                        .fill(Color(dynamicColor: tokens.backgroundColor))
+                        .frame(maxWidth: .infinity,
+                               maxHeight: .infinity,
+                               alignment: .center)
+                        .cornerRadius(tokens.cornerRadius)
+        )
+        .contentShape(Rectangle())
+        .onChange(of: isPresented, perform: { present in
+            if present {
+                presentAnimated()
+            } else {
+                dismissAnimated()
+            }
+        })
+        .onAnimationComplete(for: presentationScaleFactor) {
+            resetScaleFactor()
+        }
+        .opacity(opacity)
+        .scaleEffect(presentationScaleFactor)
+        .onTapGesture {
+            state.tapAction?()
+        }
+    }
+
+    /// UIKit wrapper that exposes the SwiftUI Heads-up Display implementation.
```

Comment:
Not sure how this is a UIKit wrapper?

Reply:
Incorrect documentation comment. Fixed it.